### ISSUE TITLE
Sanitas SBF 70 stops working (absolutely no measurements)

### ISSUE THREAD

**Issue by sebmate** [oliexdev/openScale#230]
Hi all,

first of all, thanks to all developers of openScale! It's great to see that you are "freeing" these devices!

I have some major connection issues with this scale. I had it working twice (and I'm familiar with the setup procedure), but after some time the scale stops to send any values to the smartphone (not even weight). Both devices are connected, and the bluetooth symbol in the LCD turns on and off as expected.

Resetting the scale does not work. This afternoon I had my "second" success after connecting the scale to another Android device (a Windows tablet which booted Android-x86 CM 14.1-r1) - I think this helped, but as I said, the problem returned.

I have the feeling that this could be due to other bluetooth devices interfering. Maybe openScale tries to connect to another device in my (really big) apartment building, or the firmware in the scale gets confused by this environment.

I'm using the FDroid build (1.7.1) on a Android 7.0 phone (Wilefox Spark).

I would love to further investigate this - how can I access the app's logs?

Keep up the good work!
Best, Sebastian

**Comment by Dododappere** [comment]
Difficult to say what the issue is exactly. But most likely the issue is that both devices are trying to connect to the scale and 'fight' for the connection. The scale can only have 1 active connection at a time. If one device 'takes' the connection, another device will not be able to connect to it...

So if you are standing on the scale AND you see the Bluetooth icon on, a device is connected. Find that device by turn off the devices one by one and checking again on the sale. Once all devices are off there should be no Bluetooth icon visible anymore as you stand on the scale. Then try again with the device you want to use....measurements should come in again!

**Comment by sebmate** [comment]
Hi Dododappere,

thanks for your tips. I observed this "fighting" for the connection, but this does not seem to be the problem. I can manage to create a stable connection between the phone and the scale (the scale then shows the Bluetooth symbol and openScale the symbol with the two dots on the left and right). But both do not appear to exchange data.

I'm not even sure if both devices create a permanent bond. Using the "reset" button on the scale also does not seem to affect which (openScale) device can connect to it.

I even traveled down into the basement of my 400-apartment building to make sure that there are no other Bluetooth devices. Maybe 

I don't know what's going on. If someone could tell me how I can access the logs (in the source code I can see that there's logging), I could dig deeper. I think that the phone's Bluetooth or Android is somehow messed up. It's strange that it WAS working.

Luckily I've found a temporary workaround: I've managed to permanently install Android x86 on that Windows tablet, and it does work with the scale (just fine). I can then export the data and transfer the CSV file via KDE Connect. Luckily the scale saves up to 30 measurements, so I don't have to do this every day.

I will continue to experiment with that and keep you updated. But I've had enough of that for now. :-)

Best, Sebastian

**Comment by oliexdev** [comment]
Well if you select the correct device in the openScale Bluetooth settings then openScale will only search for this MAC address, so an interfering can't be the problem.
Am I understand right you are you using two devices to connect to the scale?
Do you have the same user names set for the two devices?

**Comment by sebmate** [comment]
Hi oliexdev,

thanks for your feedback. It's good to know that the connection is bound to the MAC address. Yes, I was using two devices (the Wileyfox smartphone and the x86 Android tablet). Of course I've tried all various settings, including same user names. I've literally spent days trying. But it was only working on the tablet. Even a factory-reset of the phone did not help.

One interesting observation I've made: When the scale is connected via bluetooth and one steps on the scale, the weight values (on the LCD) are updated slower (one each second), compared to when the scale is NOT connected (there the LCD's refresh rate is much faster). So I think the scale was in fact sending the data, but the smartphone/openScale was not receiving anything or could not process it.

So I think it's problem with the Android version of this phone. I've returned the scale and got a Medisana BS444, which is working just fine (but the Sanitas is the nicer scale).

If this is a one-time issue, I guess we can close it ...

Keep up the good work!
Sebastian
